fhir-server-test/src/test/java/com/ibm/fhir/server/test/app/Main.java

```diff
@@ -0,0 +1,249 @@
+        DriverMetrics dm = new DriverMetrics();
+        driver.setMetrics(dm);
+        
+        // Inject the tenant-id into the properties so that it gets picked up by the FHIRClientImpl
+        if (tenantId != null) {
+            logger.info("Setting tenant-id = " + tenantId);
+            this.properties.setProperty("fhirclient.tenant.id", tenantId);
+        }
+        
+        TestContext context = new TestContext();
+        context.setUp(this.properties);
+        ExampleRequestProcessor erp = new ExampleRequestProcessor(context, tenantId, dm, this.readIterations);
+        driver.setProcessor(erp);
+        
+        
+        if (pool != null) {
+            // reasonable default to make sure we don't stall the thread-pool pipeline
+            if (maxInflight == 0) {
+                maxInflight = threads * 2;
+            }
+            driver.setPool(pool, maxInflight);
+        }
+        
+        // process the examples in the PERFORMANCE index (all examples < 1MB)
+        long start = System.nanoTime();
+        driver.processExamples(this.testType, Format.JSON);
+        long elapsed = (System.nanoTime() - start) / DriverMetrics.NANOS_MS;
+        if (elapsed == 0) {
+            elapsed = 1; // unlikely, but just to be on the safe side
+        }
+        renderReport(dm, elapsed);
+    }
+    
+    /**
+     * Pretty-print the response time metrics collected during the run
+     * @param dm
+     * @param elapsed number of milliseconds elapsed
+     */
+    protected void renderReport(DriverMetrics dm, long elapsed) {
+        // how many seconds did we run for?
+        double secs = elapsed / 1000.0;
+        dm.render(new DriverStats(System.out, secs));
+    }
+
+    /**
+     * Release any resources
+     */
+    protected void shutdown() {
+        if (pool != null) {
+            logger.info("Shutting down pool");
+            pool.shutdown();
+            try {
+                pool.awaitTermination(poolShutdownWaitTime, TimeUnit.MILLISECONDS);
+                logger.info("Pool shutdown complete");
+            }
+            catch (InterruptedException x) {
+                // Best effort to stop everything now
+                logger.warning("Pool shutdown interrupted. Attempting to force termination");
+                pool.shutdownNow();
+            }
+        }
+    }
+
+    /**
+     * Main entry
+     * @param args
+     */
+    public static void main(String[] args) {
+
+        Main m = new Main();
+        try {
+            m.parseArgs(args);
+            m.configure();
+            
+            long start = System.nanoTime();
+            m.process();
+            long end = System.nanoTime();
+            
+            logger.info(String.format("Took: %6.3f seconds", (end-start)/NANOS));
+        }
+        catch (Exception x) {
+            logger.log(Level.SEVERE, "Failed to run", x);
+        }
+        finally {
+            m.shutdown();
```

Comment:
we'd better add System.exit(0) after m.shutdown(), otherwise, eclipse always shows the program as not finished in it's console UI.

Reply:
done